nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/model/state/repo/AbstractWSModelRepository.java

```diff
@@ -0,0 +1,128 @@
+package com.antheminc.oss.nimbus.domain.model.state.repo;
+
+import java.io.Serializable;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import com.antheminc.oss.nimbus.FrameworkRuntimeException;
+import com.antheminc.oss.nimbus.context.BeanResolverStrategy;
+import com.antheminc.oss.nimbus.domain.model.state.repo.db.SearchCriteria;
+import com.antheminc.oss.nimbus.support.JustLogit;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter @Setter
+public abstract class AbstractWSModelRepository {
+	
+	private final BeanResolverStrategy beanResolver;
+	private final RestTemplate restTemplate;
+	private final static JustLogit logit = new JustLogit(AbstractWSModelRepository.class);
+	public AbstractWSModelRepository(BeanResolverStrategy resolver) {
+		this.beanResolver = resolver;
+		this.restTemplate = beanResolver.get(RestTemplate.class);
+	}
+
+	abstract public String getTargetUrl(String alias);
+	
+	abstract public String constructSearchUri(SearchCriteria<?> searchCriteria);
+	
+	abstract public <ID extends Serializable, T> T handleGet(ID id, Class<T> referredClass, URI url);
+	
+	abstract public <T> Object handleSearch(Class<T> referredDomainClass, Supplier<SearchCriteria<?>> criteriaSupplier, String alias, URI uri);
+	
+	protected URI createUriForAlias(String alias, String url) {
+		String targetUrl = getTargetUrl(alias);
+		if(targetUrl == null)
+			return null;
+		
+		String urlToConstruct = StringUtils.startsWith(url, "/") ? url : "/".concat(url);
+		urlToConstruct = targetUrl.concat(urlToConstruct);
+		try {
+			URI uri = new URI(urlToConstruct);
+			return uri;
+		} catch (URISyntaxException e) {
+			throw new FrameworkRuntimeException("Cannot create URI from supplied url: "+url);
+		}
+	}
+	
+	public <ID extends Serializable, T> T _get(ID id, Class<T> referredClass, String alias, String url) {
+		URI uri = createUriForAlias(alias, url);
+		if(uri == null)
+			return null;
+
+		try {
+			return handleGet(id, referredClass, uri);
+		} catch(Exception e) {
+			logit.error(() -> "Exception occured while calling Web service "+e.getMessage());
```

Comment:
No need to log error while the next line is throwing the exception out. This would create abundant error messages stacktrace repeating multiple times without any new information being logged. Would suggest to throw the new exception passing new message that includes information from the method input parameters that would help debugging.